src/main/java/uk/ac/sanger/sccp/stan/service/releasefile/ReleaseFileService.java

```diff
@@ -414,6 +427,181 @@ public void loadSectionDate(Collection<ReleaseEntry> entries, Ancestry ancestry)
         }
     }
 
+    /**
+     * Loads the various fields associated with xenium ops
+     * @param entries the release entries under construction
+     * @param slotIds the ids of slots for these entries
+     */
+    public void loadXeniumFields(Collection<ReleaseEntry> entries, Set<Integer> slotIds) {
+        loadProbeHybridisation(entries, slotIds);
+        loadProbeHybridisationQC(entries, slotIds);
+        loadXeniumAnalyser(entries, slotIds);
+        loadXeniumQC(entries, slotIds);
+    }
+
+    /**
+     * Loads timestamps and probe info for probe hybridisation ops
+     */
+    public void loadProbeHybridisation(Collection<ReleaseEntry> entries, Set<Integer> slotIds) {
+        OperationType opType = opTypeRepo.getByName("Probe hybridisation Xenium");
+        List<Operation> ops = opRepo.findAllByOperationTypeAndDestinationSlotIdIn(opType, slotIds);
+        if (ops.isEmpty()) {
+            return;
+        }
+        List<Integer> opIds = ops.stream().map(Operation::getId).collect(toList());
+        List<LabwareProbe> lwProbes = lwProbeRepo.findAllByOperationIdIn(opIds);
+        Map<Integer, List<LabwareProbe>> opIdProbes = lwProbes.stream()
+                .collect(groupingBy(LabwareProbe::getOperationId));
+        Map<Integer, Operation> labwareProbeOp = labwareIdToOp(ops);
+        for (ReleaseEntry entry : entries) {
+            final Integer lwId = entry.getLabware().getId();
+            Operation op = labwareProbeOp.get(lwId);
+            if (op==null) {
+                continue;
+            }
+            entry.setHybridStart(op.getPerformed());
+            List<LabwareProbe> probes = opIdProbes.get(op.getId());
+            if (nullOrEmpty(probes)) {
+                continue;
+            }
+            probes = probes.stream().filter(p -> p.getLabwareId().equals(lwId))
+                    .collect(toList());
+            if (!probes.isEmpty()) {
+                String plex = probes.stream()
+                        .map(LabwareProbe::getPlex)
+                        .distinct()
+                        .map(String::valueOf)
+                        .collect(joining(", "));
+                String probeName = probes.stream()
+                        .map(lp -> lp.getProbePanel().getName())
+                        .distinct()
+                        .collect(joining(", "));
+                String lot = probes.stream()
+                        .map(LabwareProbe::getLotNumber)
+                        .distinct()
+                        .collect(joining(", "));
+                entry.setXeniumPlex(plex);
+                entry.setXeniumProbe(probeName);
+                entry.setXeniumProbeLot(lot);
+            }
+        }
```

Comment:
Would be more efficient if we do something like :  
              StringJoiner plexJoiner = new StringJoiner(", ");
                StringJoiner probeNameJoiner = new StringJoiner(", ");
                StringJoiner lotJoiner = new StringJoiner(", ");

                probes.forEach(lp -> {
                    plexJoiner.add(String.valueOf(lp.getPlex()));
                    probeNameJoiner.add(lp.getProbePanel().getName());
                    lotJoiner.add(lp.getLotNumber());

Reply:
It would perform only one iteration of the `probes` list, but it's still O(N), and it's a short list. In practice there's no way it would be appreciably faster; and it isn't doing the `distinct()` steps.